sherpa-onnx/jni/jni.cc

```diff
@@ -76,11 +76,22 @@ class SherpaOnnx {
 
   bool IsReady() const { return recognizer_.IsReady(stream_.get()); }
 
-  void Reset(bool recreate) {
-    if (recreate) {
+  // If keywords is an empty string, it just recreates the decoding stream
+  // If keywords is not empty, it will create a new decoding stream with
+  // the given keywords appended to the default keywords.
+  void Reset(bool recreate, const std::string &keywords = {}) {
+    if (keywords.empty()) {
+      if (recreate) {
       stream_ = recognizer_.CreateStream();
-    } else {
+      } else {
       recognizer_.Reset(stream_.get());
+      }
+    } else {
+      auto stream = recognizer_.CreateStream(keywords);
+      // Set new keywords failed, the stream_ will not be updated.
+      if (stream != nullptr) {
+        stream_ = std::move(stream);
```

Comment:
Could you also use `SHERPA_ONNX_LOGE()` to print some logs here saying it failed to set the hotwords?